### ISSUE TITLE
Revamp the config system

### ISSUE THREAD

**Issue by nycz** [kalpana-org/kalpana#58]
Right now plugins work like crap when it comes to configs. All configs should be handled in a unified way, including reloading, updating etc.

Also more automatic if possible @ adding new settings. Not sure of specifics atm.

**Comment by nycz** [comment]
On the topic of config system shizzle, just what should the policy on error checking and verbosity due to it be for kalpana's config system? 

Examples of what to do when a setting is broken:
* It's quietly replaced by the default.
* It's detected on startup/load of config and kalpana ragequits.
* It's not detected until it's used and then everything explodes in python errors.

Ways to detect the validity of a setting:
* Duck typing, ish. When kalpana tries to use it it either works or not. Exciting!
* Static typing defined in default config (int/bool/anything as an argument for each option)
* Static typing defined in the code where the setting is _registered_.
* Static typing defined in the code when the setting is _read_. Aka it crashes at the same time as duck typing but possibly with more obvious error messages.
* Unique error checking for each setting. Should catch all errors but needs unique code for every setting _or_ a mix between static (this should be an int yo) _and_ special snowflake error checking which leads to even more code.

Also, when kalpana finds a setting that isn't used/defined by defaultconfig, should it let it be or purge it? Atm it deletes the extra setting cause I... don't know. Feels kinda pointless and needlessly coddling (and potentially dangerous if someone misspells a setting when editing).

Also not sure about the whole automatic/manual dealio... sigh. 

Anywho, at one end of the scale we've got no error checking at all (but instead a robust documentation which we should have either way) and a program that explodes if the config is b0rken, and at the other end we've got a config that is both autoupdated, autocorrected, and a kalpana that checks all settings on startup and gives helpful and useful error messages abt exactly what is incorrect and how you might fix it.

halp pls @cefyr
